### PR TITLE
Use Roslyn PublicSign, update packages.

### PR DESCRIPTION
**Pull request by AndyGerlicher**
I wanted to do more to reduce dependency on BuildTools but the changes started spiraling out of control. This will at least get rid of the signing dependency and use the Roslyn feature.

### CODE REVIEW

File: src/dir.targets [hunk 1]
```diff
@@ -35,12 +35,21 @@
   <Import Project="$(RepoRoot)\targets\OverrideGetReferenceAssemblyPaths.targets"
           Condition="'$(NetCoreBuild)' == 'true'"/>
 
-  <!-- OSS sign assemblies after compile runs. -->
-  <Import Condition="Exists('$(ToolsDir)\sign.targets')" Project="$(ToolsDir)\sign.targets" />
-  
+  <PropertyGroup Condition="'$(IsTestProject)' != 'true'">
```
Comment (cdmihai) [inline src/dir.targets]: Satellite assembly signing probably also needs to be changed: [https://github.com/Microsoft/msbuild/blob/master/src/UpdateLocalizedResources.targets#L112-L140](https://github.com/Microsoft/msbuild/blob/master/src/UpdateLocalizedResources.targets#L112-L140)

Not sure how you open source sign an existing dll with Roslyn. If it does support signing dlls, then probably you can just delete the code and register the dlls with Roslyn. The CI does debug localized

**Comment by radical** [comment]
Where is `$(UseOpenSourceSign)` being set? On mono we are forcing public signing, as Roslyn doesn't support signing on non-windows platforms currently. https://github.com/mono/msbuild/commit/665f843ac664eb2b4541242c1d6e364a9a838ac8

**Comment by rainersigwald** [comment]
@dotnet-bot test this please

**Comment by rainersigwald** [comment]
This still fails on my machine with this signing error:

```
  Unhandled Exception: System.IO.FileLoadException: Could not load file or assembly 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. Strong name validation failed. (Exception from HRESULT: 0x8013141A) ---> System.Security.SecurityExceptio
  n: Strong name validation failed. (Exception from HRESULT: 0x8013141A)
     --- End of inner exception stack trace ---
     at Microsoft.Cci.Extensions.HostEnvironment.<GetFilePaths>d__68.MoveNext()
     at Microsoft.Cci.Extensions.HostEnvironment.LoadAssemblies(IEnumerable`1 paths)
     at GenAPI.Program.Main(String[] args)
s:\msbuild\packages\Microsoft.DotNet.BuildTools.GenApi\1.0.0-beta3-01423-01\build\Microsoft.DotNet.BuildTools.GenAPI.targets(27,5): error MSB3073: The command ""s:\msbuild\packages\Microsoft.DotNet.BuildTools.GenApi\1.0.0-beta3-01423-01\build\\..\tools\GenApi.exe" -assembly:"s:\msbuild\bin\x86\Windows_NT\D
ebug\Output\Microsoft.Build.Framework.dll" -libPath:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6" -out:"s:\msbuild\ref\net46\Microsoft.Build.Framework\Microsoft.Build.Framework.cs" " exited with code -532462766. [s:\msbuild\src\Framework\Microsoft.Build.Framework.cspr
oj]
```

But I fixed some of it! (thanks for the pointer @radical!)

**Comment by AndyGerlicher** [comment]
@dotnet-bot test mono please

**Comment by AndyGerlicher** [comment]
@dotnet-bot test mono please

**Comment by AndyGerlicher** [comment]
@dotnet-bot test mono please

**Comment by rainersigwald** [comment]
@AndyGerlicher I'm still seeing this failure on my machine (now in master :()

```
  Unhandled Exception: System.IO.FileLoadException: Could not load file or assembly 'System.IO.FileSystem, Version=4.0.1.0, Culture=neutral, Publi
  cKeyToken=b03f5f7f11d50a3a' or one of its dependencies. Strong name validation failed. (Exception from HRESULT: 0x8013141A) ---> System.Security
  .SecurityException: Strong name validation failed. (Exception from HRESULT: 0x8013141A)
     --- End of inner exception stack trace ---
     at Microsoft.Cci.Extensions.HostEnvironment.<GetFilePaths>d__68.MoveNext()
     at Microsoft.Cci.Extensions.HostEnvironment.LoadAssemblies(IEnumerable`1 paths)
     at GenAPI.Program.Main(String[] args)
  MSBuildTaskHost -> s:\msbuild\bin\Debug\x86\Windows_NT\Output\MSBuildTaskHost.exe
s:\msbuild\packages\Microsoft.DotNet.BuildTools.GenApi\1.0.0-beta3-01902-02\build\Microsoft.DotNet.BuildTools.GenAPI.targets(30,5): error MSB3073:
 The command ""s:\msbuild\packages\Microsoft.DotNet.BuildTools.GenApi\1.0.0-beta3-01902-02\build\\..\tools\GenAPI.exe" -assembly:"s:\msbuild\bin\D
ebug\x86\Windows_NT\Output\Microsoft.Build.Framework.dll" -libPath:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\
v4.6" -out:"s:\msbuild\ref\net46\Microsoft.Build.Framework\Microsoft.Build.Framework.cs" " exited with code -532462766. [s:\msbuild\src\Framework\
Microsoft.Build.Framework.csproj]
```

**Pull request merged**